<div align="center">

**LAPORAN PRAKTIKUM**

**STRUKTUR DATA**

**“Tipe Data dan Variabel”**

**Disusun Oleh :**

**Adib Ridho Putra**

**2611532018**

**Dosen Pengampu:**

**Dr. Ir. Wahyudi, S.T., M.T.**

**Asisten Praktikum:**

**Rifky Yuliandra**

<br>

 <img src="images/logo_unand.png" width="200">

<br>

**FAKULTAS TEKNOLOGI INFORMASI**

**DEPARTEMEN INFORMATIKA**

**UNIVERSITAS ANDALAS**

**2026**

</div>

## KATA PENGANTAR

Puji syukur penulis panjatkan kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan karunia-Nya, sehingga penulis dapat menyelesaikan laporan praktikum “Tipe Data dan Variabel” yang telah dilaksanakan pada praktikum pekan ke-2 untuk mata kuliah Praktikum Struktur Data. Praktikum ini berfokus pada pembahasan mengenai konsep tipe data dan variabel dalam pemrograman, meliputi boolean, input, konstanta, numerik, dan string.

Melalui praktikum ini, penulis diharapkan dapat memahami konsep dasar tipe data dan variabel serta mampu mengimplementasikannya dalam program sederhana. Selain itu, praktikum ini diharapkan dapat memberikan pemahaman mengenai penggunaan berbagai jenis data dan pengelolaannya dalam program sehingga dapat menjadi dasar dalam mempelajari materi pemrograman selanjutnya.

Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan laporan ini di masa mendatang.

Padang, 2026

---
# BAB I
# PENDAHULUAN

## 1.1 Latar Belakang
Pemrograman merupakan salah satu bagian penting dalam bidang ilmu komputer yang digunakan untuk memberikan instruksi kepada komputer agar dapat melakukan suatu proses sesuai dengan kebutuhan pengguna. Dalam proses pemrograman, data menjadi salah satu unsur utama yang harus dikelola dengan baik. Setiap data memiliki karakteristik dan bentuk yang berbeda sehingga diperlukan pemahaman mengenai cara menyimpan dan mengolah data tersebut di dalam sebuah program.

Salah satu konsep dasar yang perlu dipahami dalam pemrograman adalah tipe data dan variabel. Tipe data digunakan untuk menunjukkan jenis atau karakteristik suatu nilai yang akan digunakan dalam program, sedangkan variabel digunakan sebagai tempat untuk menyimpan nilai tersebut. Pemahaman terhadap kedua konsep ini penting karena hampir setiap program membutuhkan proses penyimpanan, perubahan, dan pengolahan data.

Dalam pemrograman Python, terdapat berbagai bentuk data yang dapat digunakan, di antaranya boolean, numerik, dan string. Boolean digunakan untuk merepresentasikan nilai benar atau salah, numerik digunakan untuk mengolah data berupa angka, sedangkan string digunakan untuk menyimpan data berupa teks. Selain memahami jenis data tersebut, penggunaan input juga diperlukan agar program dapat menerima data dari pengguna. Sementara itu, konstanta digunakan untuk menyimpan nilai yang ditetapkan tetap dan tidak dimaksudkan untuk berubah selama program dijalankan.

Pemahaman mengenai tipe data dan variabel menjadi dasar dalam pembuatan program karena kesalahan dalam menentukan atau menggunakan data dapat memengaruhi proses pengolahan program. Oleh sebab itu, melalui praktikum pekan ke-2 ini, mahasiswa mempelajari konsep tipe data dan variabel serta penerapannya dalam program menggunakan Python. Praktikum ini diharapkan dapat membantu mahasiswa memahami bagaimana data didefinisikan, disimpan, dimasukkan, dan digunakan dalam suatu program sebagai dasar untuk mempelajari konsep pemrograman yang lebih kompleks.

## 1.2 Tujuan
1. Memahami konsep dasar tipe data dan variabel dalam pemrograman.
2. Mengetahui penggunaan tipe data boolean, numerik, dan string dalam Python.
3. Memahami penggunaan input untuk menerima data dari pengguna.
4. Mampu membuat dan menggunakan variabel dalam program Python.
5. Mampu menerapkan berbagai jenis data dan variabel dalam program sederhana.

## 1.3 Manfaat
1. Menambah pemahaman mengenai konsep dasar pemrograman, khususnya tipe data dan variabel.
2. Meningkatkan kemampuan dalam menentukan dan menggunakan tipe data yang sesuai dengan kebutuhan program.
3. Melatih kemampuan dalam membuat serta mengelola variabel pada Python.

---
# BAB II
# PEMBAHASAN

## 2.1 Dasar Teori
Tipe data dan variabel merupakan konsep dasar dalam pemrograman yang digunakan untuk menyimpan dan mengelola data. Tipe data menunjukkan jenis data yang digunakan dalam suatu program, sedangkan variabel digunakan sebagai tempat untuk menyimpan nilai yang dapat digunakan dalam proses pemrograman. Pemahaman mengenai tipe data dan variabel diperlukan agar data dapat diproses sesuai dengan karakteristik dan kebutuhan program.
1. **Boolean**
merupakan jenis data yang memiliki dua nilai, yaitu True dan False. Boolean digunakan untuk menyatakan suatu kondisi yang hanya memiliki dua kemungkinan, seperti benar atau salah. Dalam pemrograman, boolean banyak digunakan dalam pemeriksaan kondisi dan proses pengambilan keputusan.
2. **Input**
merupakan proses memasukkan atau menerima data dari pengguna ke dalam program. Data yang diberikan melalui input dapat disimpan ke dalam variabel dan selanjutnya digunakan dalam proses pengolahan data. Input memungkinkan program menerima informasi secara langsung dari pengguna.
3. **Konstanta**
merupakan nilai yang ditetapkan dalam program dan digunakan sebagai nilai tetap. Berbeda dengan variabel yang nilainya dapat berubah, konstanta digunakan ketika suatu nilai perlu dipertahankan agar tetap sama selama program digunakan. Konstanta dapat membantu menjaga konsistensi nilai tertentu dalam suatu program.
4. **Numerik**
merupakan data yang berupa angka dan digunakan untuk melakukan berbagai operasi perhitungan. Data numerik dapat digunakan dalam operasi matematika seperti penjumlahan, pengurangan, perkalian, dan pembagian. Data numerik menjadi salah satu jenis data penting dalam program yang melakukan pengolahan angka.
5. **String**
merupakan data yang digunakan untuk menyimpan teks atau kumpulan karakter. String dapat berupa huruf, kata, kalimat, angka yang diperlakukan sebagai teks, maupun karakter tertentu. Dalam pemrograman, string umumnya digunakan untuk menyimpan data berbentuk teks yang perlu ditampilkan atau diolah oleh program.

## 2.2 Kode Program

### 2.2.1 Program "Boolean_2611532018"

In [34]:
# Buat file dengan nama Boolean_NIM.py
# nama variabel ditambah 4 digit nim terakhir contoh: nilai_1234
# Deklarasi variabel dengan tipe data Boolean
is_lulus_2018 = True
is_cumlaude_2018 = True

# Menggunakan Boolean
nilai_2018 = 85
batas_lulus_2018 = 75

# Menentukan nilai Boolean dari kondisi
status_kelulusan_2018 = nilai_2018 >= batas_lulus_2018 # Hasilnya akan True

print("=== Check Kelulusan ===")
print("Nilai:", nilai_2018)
print("Apakah Lulus?:", status_kelulusan_2018)
if is_lulus_2018 and is_cumlaude_2018:
    print("Selamat, Anda Lulus dengan Predikat Cum Laude!")

=== Check Kelulusan ===
Nilai: 85
Apakah Lulus?: True
Selamat, Anda Lulus dengan Predikat Cum Laude!


**Penjelasan :**
1. Baris 4, **is_lulus_2018 = True** berfungsi membuat variabel is_lulus_2018 dengan nilai True.
2. Baris 5, **is_cumlaude_2018 = True** berfungsi membuat variabel is_cumlaude_2018 dengan nilai True.
3. Baris 8, **nilai_2018 = 85** berfungsi menyimpan nilai mahasiswa.
4. Baris 9, **batas_lulus_2018 = 75** berfungsi menentukan nilai minimum untuk lulus.
5. Baris 12, **status_kelulusan_2018 = nilai_2018 >= batas_lulus_2018** berfungsi membandingkan nilai dengan batas lulus.
6. Baris 14, **print("=== Check Kelulusan ===")** berfungsi menampilkan tulisan Check Kelulusan sebagai judul.
7. Baris 15, **print("Nilai:", nilai_2018)** berfungsi menampilkan nilai mahasiswa, yaitu 85.
8. Baris 16, **print("Apakah Lulus?:", status_kelulusan_2018)** berfungsi menampilkan hasil pengecekan kelulusan.
9. Baris 17, **if is_lulus_2018 and is_cumlaude_2018:** Mengecek apakah kedua kondisi bernilai True.
and berarti kedua kondisi harus benar.
10. Baris 18, **print("Selamat, Anda Lulus dengan Predikat Cum Laude!")** Jika kedua kondisi benar, program menampilkan ucapan selamat, anda lulus dengan predikat Cum Laude.

### 2.2.2 Program "Input_2611532018"

In [ ]:
# Buat file dengan nama Input_NIM.py
# Buat program untuk menghitung luas persegi panjang
# Nama variabel ditambah 4 digit nim terakhir contoh: panjang_1234
# Program ini menggunakan fungsi input() untuk meminta pengguna
# memasukkan nilai panjang dan lebar persegi panjang.
# Nilai yang dimasukkan akan dikonversi menjadi tipe data integer
panjang_2018 = input('Masukan nilai panjang: ')
lebar_2018 = input('Masukkan nilai lebar: ')
luas_2018 = int(panjang_2018) * int(lebar_2018)
print("Luas:", luas_2018)

**Penjelasan :**
1. Baris 7, **panjang_2018 = input('Masukan nilai panjang: ')** berfungsi untuk meminta pengguna untuk memasukkan nilai panjang. Nilai tersebut disimpan dalam variabel panjang_2018
2. Baris 8: **lebar_2018 = input('Masukkan nilai lebar: ')** berfungsi meminta pengguna untuk memasukkan nilai lebar. Nilai tersebut disimpan dalam variabel lebar_2018.
3. Baris 9, **luas_2018 = int(panjang_2018) * int(lebar_2018)** berfungsi mengubah nilai panjang dan lebar menjadi integer menggunakan int(). Kemudian mengalikan panjang dengan lebar untuk mendapatkan luas persegi panjang. Rumusnya: Luas = panjang × lebar.
4. Baris 10, **print("Luas:", luas_2018)** berfungsi menampilkan hasil perhitungan luas ke layar.

### 2.2.3 Program "Konstanta_2611532018"

In [ ]:
# Buat file dengan nama Konstanta_NIM.py
# Program ini menggunakan konstanta untuk menghitung luas lingkaran
# nama variabel ditambah 4 digit nim terakhir contoh: jari_1234

from typing import Final
PI: Final = 3.14
print("pi: %f" % (PI))
jari_2018 = float(input('masukkan nilai jari-jari: '))
luas_2018 = PI * jari_2018 * jari_2018
print("Luas lingkaran dengan jari-jari %.2f adalah %.2f" % (jari_2018, luas_2018))

pi: 3.140000
Luas lingkaran dengan jari-jari 10.00 adalah 314.00


**Penjelasan :**
1. Baris 5, **from typing import Final** berfungsi untuk mengambil Final dari Python. Final digunakan untuk menandai bahwa suatu nilai dianggap sebagai konstanta dan tidak seharusnya diubah.
2. Baris 6, **PI: Final = 3.14** berfungsi untuk membuat konstanta PI dengan nilai 3.14.
Nilai ini digunakan dalam rumus luas lingkaran.
3. Baris 7, **print("pi: %f" % (PI))** berfungsi untuk menampilkan nilai PI ke layar. %f digunakan untuk menampilkan nilai dalam bentuk angka desimal.
4. Baris 8, **jari_2018 = float(input('masukkan nilai jari-jari: '))** berfungsi untuk meminta pengguna memasukkan nilai jari-jari lingkaran. input() digunakan untuk menerima masukan dari pengguna. float() digunakan agar nilai yang dimasukkan dapat berupa angka desimal. Nilai tersebut disimpan dalam variabel jari_2018.
5. Baris 9, **luas_2018 = PI * jari_2018 * jari_2018** berfungsi untuk menghitung luas lingkaran menggunakan rumus: Luas = π × r × r. Hasil perhitungan disimpan dalam variabel luas_2018.
6. Baris 10, **print("Luas lingkaran dengan jari-jari %.2f adalah %.2f" % (jari_2018, luas_2018))** berfungsi untuk menampilkan hasil perhitungan luas lingkaran. %.2f digunakan untuk menampilkan angka dengan 2 angka di belakang koma.

### 2.2.4 Program "Numerik_2611532018"

In [ ]:
# Buat file dengan nama Numerik_NIM.py
# nama variabel ditambah 4 digit terakhir NIM contoh: a_5500
#create a variabel with numeric type value.
a_2018=100
print("The type of variable having value", a_2018, "is ", type(a_2018))

#create a variable with float value.
b_2018=10.2345
print("The type of variable having value", b_2018, "is ", type(b_2018))

#create a variable with complex value.
c_2018=100+3j
print("The type of variable having value", c_2018, "is ", type(c_2018))

The type of variable having value 100 is  <class 'int'>
The type of variable having value 10.2345 is  <class 'float'>
The type of variable having value (100+3j) is  <class 'complex'>


**Penjelasan :**
1. Baris 4 dan 5, **a_2018=100** membuat variabel a_2018 dengan nilai 100, sedangkan **print("The type of variable having value", a_2018, "is ", type(a_2018))** menampilkan nilai dari a_2018 dan mengecek tipe datanya.
3. Baris 8 dan 9, **b_2018=10.2345** membuat variabel b_2018 dengan nilai 10.234, sedangkan **print("The type of variable having value", b_2018, "is ", type(b_2018))** menampilkan nilai b_2018 dan mengecek tipe datanya.
5. Baris 12 dan 13, **c_2018=100+3j** membuat variabel c_2018 dengan nilai 100 + 3j, sedangkan **print("The type of variable having value", c_2018, "is ", type(c_2018))** menampilkan nilai c_2018 dan mengecek tipe datanya.

### 2.2.5 Program "String_2611532018"

In [ ]:
# Buat file dengan nama String_NIM.py
#nama variabel ditambah 4 digit nim terakhir contoh_1234
#ganti programnya dan data masing2
nama_2018 = "Adib"
jenis_kelamin_2018 = 'L'
alamat_2018 = """
   Nagari Pauah
   Kec. Lubuk Sikaping
   Kab. Pasaman
"""
agama_2018 = 'Islam'
print("Nama saya adalah: ", nama_2018)
print("Jenis kelamin saya adalah: ", jenis_kelamin_2018)
print("Alamat saya adalah: ", alamat_2018)
print("Agama saya adalah: ", agama_2018)

Nama saya adalah:  Adib
Jenis kelamin saya adalah:  L
Alamat saya adalah:  
   Nagari Pauah
   Kec. Lubuk Sikaping
   Kab. Pasaman

Agama saya adalah:  Islam


**Penjelasan :**
1. Baris 4 dan 12, **nama_2018 = "Adib"** membuat variabel nama_2018 untuk menyimpan nama, sedangkan **print("Nama saya adalah: ", nama_2018)** menampilkan nama yang tersimpan dalam variabel nama_2018.
2. Baris 5 dan 13, **jenis_kelamin_2018 = 'L'** membuat variabel jenis_kelamin_2018 untuk menyimpan jenis kelamin, sedangkan  **print("Jenis kelamin saya adalah: ", jenis_kelamin_2018)** menampilkan jenis kelamin yang tersimpan dalam variabel jenis_kelamin_2018.
3. Baris 6 sampai 10 dan baris 14, **alamat_2018 = """ ... """** membuat variabel alamat_2018 untuk menyimpan alamat dan tanda """ digunakan untuk menulis string yang terdiri dari beberapa baris, sedangkan **print("Alamat saya adalah: ", alamat_2018)** menampilkan alamat yang tersimpan dalam variabel alamat_2018.
4. Baris 11 dan 15, **agama_2018 = 'Islam'** membuat variabel agama_2018 untuk menyimpan agama, sedangkan **print("Agama saya adalah: ", agama_2018)** menampilkan agama yang tersimpan dalam variabel agama_2018.

---
# BAB III
# KESIMPULAN

1. Praktikum pekan ke-2 memberikan pemahaman mengenai konsep dasar tipe data dan variabel dalam pemrograman menggunakan Python.
2. Tipe data boolean, numerik, dan string memiliki karakteristik dan fungsi yang berbeda sesuai dengan jenis data yang disimpan atau diolah.
3. Variabel digunakan sebagai tempat untuk menyimpan data sehingga nilai tersebut dapat digunakan dan diolah dalam program.

---
## DAFTAR PUSTAKA

[1] Python Software Foundation. "Built-in Types — Python 3 Documentation".
https://docs.python.org/3/library/stdtypes.html. 

[2] Python Software Foundation. "Built-in Functions — Python 3 Documentation".
https://docs.python.org/3/library/functions.html

[3] Python Software Foundation. "The Python Tutorial — An Informal Introduction to Python".
https://docs.python.org/3/tutorial/